libs/wingsdk/src/target-tf-aws/website.ts

```diff
@@ -0,0 +1,125 @@
+import { readdirSync } from "fs";
+import path from "path";
+import { CloudfrontDistribution } from "@cdktf/provider-aws/lib/cloudfront-distribution";
+import { S3Bucket } from "@cdktf/provider-aws/lib/s3-bucket";
+
+import { S3BucketWebsiteConfiguration } from "@cdktf/provider-aws/lib/s3-bucket-website-configuration";
+import { S3Object } from "@cdktf/provider-aws/lib/s3-object";
+import { Construct } from "constructs";
+import mime from "mime-types";
+import { createEncryptedBucket } from "./bucket";
+import { core } from "..";
+import * as cloud from "../cloud";
+import { Json } from "../std";
+
+const INDEX_FILE = "index.html";
+
+/**
+ * AWS implementation of `cloud.Website`.
+ *
+ * @inflight `@winglang/sdk.cloud.IWebsiteClient`
+ */
+export class Website extends cloud.Website {
+  private readonly bucket: S3Bucket;
+
+  constructor(scope: Construct, id: string, props: cloud.WebsiteProps) {
+    super(scope, id, props);
+
+    this.bucket = createEncryptedBucket(this, true, "website-bucket");
+
+    new S3BucketWebsiteConfiguration(this, "bucket-website-configuration", {
+      bucket: this.bucket.bucket,
+      indexDocument: { suffix: INDEX_FILE },
+    });
+
+    this.uploadFiles(this.path);
+
+    // create a cloudFront distribution
+    const distribution = new CloudfrontDistribution(this, "distribution", {
+      enabled: true,
+      ...(this._domain && { aliases: [this._domain] }),
+      origin: [
+        {
+          domainName: this.bucket.bucketRegionalDomainName,
+          originId: "s3Origin",
+        },
+      ],
+      defaultRootObject: INDEX_FILE,
+      defaultCacheBehavior: {
+        allowedMethods: ["GET", "HEAD"],
+        cachedMethods: ["GET", "HEAD"],
+        targetOriginId: "s3Origin",
+        forwardedValues: {
+          queryString: false,
+          cookies: { forward: "none" },
+        },
+        compress: true,
+        viewerProtocolPolicy: "redirect-to-https",
+        minTtl: 0,
+        defaultTtl: 3600,
+        maxTtl: 86400,
+      },
+      restrictions: {
+        geoRestriction: {
+          locations: [],
+          restrictionType: "none",
+        },
+      },
+      priceClass: "PriceClass_100",
+      viewerCertificate: { cloudfrontDefaultCertificate: true },
+    });
+
+    // //TODO: this is currently a cdk token. we need to see if we can resolve those tokens somehow
+    this._url = distribution.domainName;
+  }
+
+  public addJson(filePath: string, obj: Json): string {
+    if (!filePath.endsWith(".json")) {
+      throw new Error(
+        `key must have a .json suffix: ${filePath.split(".").pop()}`
+      );
+    }
+
+    new S3Object(this, `aws_s3_bucket_object_${filePath}`, {
```

Comment:
I don't think including the name of the terraform resource is needed / usually that info is redundant
```suggestion
    new S3Object(this, `File-${filePath}`, {
```